### PR TITLE
Getter/Setter

### PR DESCRIPTION
**Pull request by Arikad0**
This is my first open source contribution to a project, hope it could be helpful, anything that I can improve on this PR, please contact me, any tips would be appreciated. This PR references to #24617.

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -2238,6 +2238,34 @@ def _redo_transform_rel_fig(self, bbox=None):
         self.bbox_relative.p0 = (x0, y0)
         self.bbox_relative.p1 = (x0 + dx, y0 + dy)
 
+    
+    def get_subplotparams(self):
+            return self.subplotpars
+
+    def set_subplot_params(self, subplot_params=None):
```
Comment (ksunden) [inline lib/matplotlib/figure.py:2245]: The getter and setter are inconsistent in `_subplotparams` or `_subplot_params`

Additionally, while I think the more spelled out name is _better_, the idea is I think to match the variable as in `__init__`, so they should probably be `[gs]et_subplotpars`, that or have the canonical as `[gs]et_subplotparams` and use `alias` to make `[gs]et_subplotpars` work as well.

I will note that the issue had the more spelled out name, admittedly.

File: lib/matplotlib/tests/test_figure.py [hunk 2]
```diff
@@ -1456,6 +1456,43 @@ def test_subfigure_pdf():
...
+
+
+def test_fig_get_set():
+    varnames = filter(lambda var: var not in ['self', 'kwargs', 'args'],
+                      Figure.__init__.__code__.co_varnames)
+    fig = plt.figure()
+    for var in varnames:
+        # if getattr fails then the getter and setter does not exist
+        getfunc = getattr(fig, f"get_{var}")
+        setfunc = getattr(fig, f"set_{var}")
```
Comment (ksunden) [inline lib/matplotlib/tests/test_figure.py:1495]: This test assumes all of #24617 is complete, which this PR as it stands does not accomplish, and so it is failing.

File: lib/matplotlib/tests/test_figure.py [hunk 3]
```diff
@@ -1456,6 +1456,43 @@ def test_subfigure_pdf():
...
+                      " this key was ignored"):
+        fig.set_subplotparams(test_dict)
+
+    with pytest.raises(TypeError,
+                       match="subplotparams must be a dictionary of "
+                       "keyword-argument pairs or "
+                       "an instance of SubplotParams()"):
+        fig.set_subplotparams(['foo'])
+
+    assert fig.subplotpars == fig.get_subplotparams()
```
Comment (ksunden) [inline lib/matplotlib/tests/test_figure.py:1485]: This test is using `set_subplotparams` which is currently implemented as `set_subplot_params`, and so it is failing.

**Review by ksunden — CHANGES_REQUESTED** [review]

**Comment by melissawm** [comment]
Hi @Arikad0 - did you mean to close this?

**Pull request closed without merging**